### ISSUE TITLE
Parse Error from Ignored File

### ISSUE THREAD

**Issue by merlinpatt** [benmosher/eslint-plugin-import#1140]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **Node Version:** 8.9.2
* **npm Version:** 6.1.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
  "eslintIgnore": ["modules/validate.js"],
  "eslintConfig": {
    "parser": "babel-eslint",
    "parserOptions": {
      "ecmaFeatures": {
        "jsx": true
      }
    },
    "plugins": [
      "babel",
      "meteor",
      "react"
    ],
    "extends": [
      "airbnb",
      "plugin:meteor/recommended",
      "plugin:react/recommended"
    ],
    "env": {
      "browser": true
    },
    "globals": {
      "expect": false
    },
    "rules": {
      "import/no-unresolved": 0,
      "import/no-extraneous-dependencies": 0,
      "import/extensions": 0,
      "jsx-a11y/anchor-is-valid": [
        "error",
        {
          "components": [
            "Link"
          ],
          "specialLink": [
            "to"
          ]
        }
      ],
      "max-len": 0,
      "no-alert": 0,
      "no-console": 0,
      "no-restricted-globals": 0,
      "no-return-assign": [
        "error",
        "except-parens"
      ],
      "no-underscore-dangle": [
        "error",
        {
          "allow": [
            "_id",
            "_ensureIndex",
            "_verifyEmailToken",
            "_resetPasswordToken",
            "_name"
          ]
        }
      ],
      "class-methods-use-this": 0,
      "react/jsx-filename-extension": 0,
      "react/forbid-prop-types": 0
    },
    "settings": {
      "import/resolver": "meteor"
    }
  }
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// validate.js, the root of the issue
import { Meteor } from 'meteor/meteor';

if (Meteor.isClient) {
  import $ from 'jquery';
  import 'jquery-validation';

  export default (form, options) => $(form).validate(options);
}

// any other file importing validate (may be a different path
import validate from '../validate';
// ...
```

```bash
eslint .
```

**What did you expect to happen?**

I have the `validate.js` file in `eslintIgnore` so it should ignore any errors from the `validate.js` file.

**What actually happened? Please include the actual, raw output from ESLint.**

On any file that imports `validate.js`, I get "parse errors in imported module."

```
/Users/merlin/code/clever-beagle/pup/ui/components/AdminUserProfile/AdminUserProfile.js
  12:22  error  Parse errors in imported module '../../../modules/validate': Line 4: 'import' and 'export' may only appear at the top level
```

Copied from [eslint #10615](https://github.com/eslint/eslint/issues/10615)

**Comment by ljharb** [comment]
I’m not sure that this is a bug; presumably if you add the file to the import/ignore setting, it would be fine. Adding a file to eslintignore tells eslint not to run rules on it; however, this plugin has to parse imported things so that it can lint the file doing the importing.

Is the only reason you’re ignoring the file because of the nonstandard meteor syntax of non-top-level imports/exports? Does meteor not provide a babel transform or an eslint parser that can cover that file?

**Comment by ljharb** [comment]
That’s in fact the most common - when you don’t want to lint a vendored file, because you didn’t write it, but you still want to ensure you’re importing something from it that exists.
